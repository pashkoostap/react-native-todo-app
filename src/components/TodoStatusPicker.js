import React from "react";
import { Picker } from "react-native";

const TodoStatusPicker = ({ input: { onChange, value } }) => (
  <Picker onValueChange={onChange} selectedValue={value}>
    <Picker.Item label="Actual" value="actual" />
    <Picker.Item label="Completed" value="completed" />
  </Picker>
);

export default TodoStatusPicker;
