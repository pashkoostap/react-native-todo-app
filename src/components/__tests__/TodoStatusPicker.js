import React from "react";
import renderer from "react-test-renderer";

import TodoStatusPicker from "../TodoStatusPicker";

describe("TodoStatusPicker", () => {
  it("should match the snapshot", () => {
    const props = { input: { onChange: () => {}, value: "" } };
    const rendered = renderer.create(<TodoStatusPicker {...props} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
