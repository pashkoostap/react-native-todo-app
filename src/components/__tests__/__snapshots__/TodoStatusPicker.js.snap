// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoStatusPicker should match the snapshot 1`] = `
<View
  style={undefined}
>
  <RCTPicker
    items={
      Array [
        Object {
          "label": "Actual",
          "textColor": undefined,
          "value": "actual",
        },
        Object {
          "label": "Completed",
          "textColor": undefined,
          "value": "completed",
        },
      ]
    }
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
